src/microsAuthentication/content.md

```diff
@@ -0,0 +1,1075 @@
+minikube start \
+  --extra-config=apiserver.service-account-signing-key-file=\
+    /var/lib/minikube/certs/sa.key \
+  --extra-config=apiserver.service-account-issuer=\
+    kubernetes/serviceaccount \
+  --extra-config=apiserver.service-account-api-audiences=api
+```
+
+You should also clone the repository <https://github.com/amitsaha/kubernetes-sa-volume-demo> as it contains the demo source code that will be referred to in the article.
+
+## Deploying the API component
+
+The API service is a headless web application listening on port 8080.
+
+When a client makes any request to it, the API component:
+
+1. Reads the service account token.
+1. Makes a HTTP GET request to the secret store service with the token linked to the Service Account.
+1. Forwards the response.
+
+You can deploy the app in the cluster with:
+
+```terminal|command=1|title=bash
+kubectl apply -f service_accounts/api/deployment.yaml
+namespace/api created
+serviceaccount/api created
+deployment.apps/app created
+```
+
+You can temporarily expose the API with:
+
+```terminal|command=1|title=bash
+kubectl -n api expose deployment
+service/app exposed
+```
+
+Retrieve the URL of the app with:
+
+```terminal|command=1|title=bash
+minikube service deployment --url
+TODO: missing output
+```
+
+_Will it work?_
+
+You can issue a request with:
+
+```terminal|command=1|title=bash
+curl <insert URL from above>
+Get "<http://app.secret-store.svc.cluster.local">: dial tcp: lookup app.secret-store.svc.cluster.local: no such host
+```
+
+This is expected since you haven't yet deployed the secret store service.
+
+_Keep the terminal open._
+
+Open a new terminal to carry out the next set of steps.
+
+## Deploying the Secret store
+
+The secret store service is another headless web application listening on port 8081.
+
+When a client makes any request to it, the Secret store:
+
+1. Looks for a token in the request. If there isn't one, it replies with a HTTP 401 error response.
+1. Checks the token with the Kubernetes API for its validity. If it's invalid, it replies with a HTTP 403 response.
+1. Replies to the original request.
+
+You can create the secret store with:
+
+```terminal|command=1|title=bash
+kubectl apply -f service_accounts/secret-store/deployment.yaml
+TODO: missing output
+```
+
+Now, go back to the previous terminal window with the API service and issue the same request again:
+
+```terminal|command=1|title=bash
+curl <insert URL from above>
+Hello from secret store. You have been authenticated
+```
+
+The secret store service successfully verified the token and replied to your request.
+
+_But how does all of that work? Let's find out._
+
+Keep the two terminal sessions running and switch to a new (third) terminal session.
+
+## Under the hood
+
+**Service accounts are a way to associate your Kubernetes workload with an identity.**
+
+You can use Service Accounts to define what resources should be accessible in the cluster and who can access them.
```

Comment:
Technically, a role grants permissions. SAs are identities in the cluster.
I think this is not correct?